src/helpers.ts

```diff
@@ -4,41 +4,39 @@
  *--------------------------------------------------------------------------------------------*/
 import * as fs from "fs";
 import * as JSONC from "jsonc-parser";
+import Mustache from "mustache";
 import * as log from "./Log";
+import ISettingsConfigJson from "./types/ISettingsConfigJson";
 
-/**
- * Loads a settings file and validates it with JSON schema, then returns it as a typed object.
- * @param serviceName The name of the service loading the settings. Used in log messages.
- * @param settingsFileName The path to the file to load.
- * @type T The type the settings should return as.
- */
-export function readSettings<T>(serviceName: string, settingsFileName: string): T {
-  let rawConfig: string;
+function readFile(filePath: string, fileType: string, serviceName: string): string {
   try {
-    rawConfig = fs.readFileSync(settingsFileName, "utf-8");
+    return fs.readFileSync(filePath, "utf-8");
   } catch (e) {
-    log.warn(serviceName, `Unable to read the configuration file: ${e.message}.`);
+    log.warn(serviceName, `Unable to read the ${fileType} file: ${e.message}.`);
     return null;
   }
+}
 
-  // This shouldn't happen. Keeping the check here in case it does in the real world
-  // and someone reports things not working.
-  if (!rawConfig) {
-    throw new Error(`[${serviceName}] Unable to load configuration file ${settingsFileName}.`);
-  }
-
-  const parseErrors: JSONC.ParseError[] = [];
+function parseFile(filePath: string, fileType: string, serviceName: string) {
+  const file = readFile(filePath, fileType, serviceName);
+  return file ? JSONC.parse(file) : null;
+}
 
-  const settings = JSONC.parse(rawConfig, parseErrors) as T;
+function replaceSecrets(settings: ISettingsConfigJson, secrets: string) {
+  return JSON.parse(Mustache.render(JSON.stringify(settings), secrets));
```

Comment:
I think this needs to be `JSONC.parse()` for consistency with the rest of the code. Otherwise there will be one situation where you can't use JSON comments in the input file.

Reply:
Updated.